sdks/python/apache_beam/io/vcfio.py

```diff
@@ -0,0 +1,440 @@
+  """
+
+  def __init__(self,
+               reference_name=None,
+               start=None,
+               end=None,
+               reference_bases=None,
+               alternate_bases=None,
+               names=None,
+               quality=None,
+               filters=None,
+               info=None,
+               calls=None):
+    """Initialize the :class:`Variant` object.
+
+    Args:
+      reference_name (str): The reference on which this variant occurs
+        (such as `chr20` or `X`). .
+      start (int): The position at which this variant occurs (0-based).
+        Corresponds to the first base of the string of reference bases.
+      end (int): The end position (0-based) of this variant. Corresponds to the
+        first base after the last base in the reference allele.
+      reference_bases (str): The reference bases for this variant.
+      alternate_bases (List[str]): The bases that appear instead of the
+        reference bases.
+      names (List[str]): Names for the variant, for example a RefSNP ID.
+      quality (float): Phred-scaled quality score (-10log10 prob(call is wrong))
+        Higher values imply better quality.
+      filters (List[str]): A list of filters (normally quality filters) this
+        variant has failed. `PASS` indicates this variant has passed all
+        filters.
+      info (dict): A map of additional variant information. The key is specified
+        in the VCF record and the value is of type ``VariantInfo``.
+      calls (list of :class:`VariantCall`): The variant calls for this variant.
+        Each one represents the determination of genotype with respect to this
+        variant.
+    """
+    self.reference_name = reference_name
+    self.start = start
+    self.end = end
+    self.reference_bases = reference_bases
+    self.alternate_bases = alternate_bases or []
+    self.names = names or []
+    self.quality = quality
+    self.filters = filters or []
+    self.info = info or {}
+    self.calls = calls or []
+
+  def __eq__(self, other):
+    return (isinstance(other, Variant) and
+            vars(self) == vars(other))
+
+  def __repr__(self):
+    return ', '.join(
+        [str(s) for s in [self.reference_name,
+                          self.start,
+                          self.end,
+                          self.reference_bases,
+                          self.alternate_bases,
+                          self.names,
+                          self.quality,
+                          self.filters,
+                          self.info,
+                          self.calls]])
+
+  def __lt__(self, other):
+    if not isinstance(other, Variant):
+      return NotImplemented
+
+    if self.reference_name != other.reference_name:
+      return self.reference_name < other.reference_name
+    elif self.start != other.start:
+      return self.start < other.start
+    elif self.end != other.end:
+      return self.end < other.end
+
+    self_vars = vars(self)
```

Comment:
Add a comment why are the 3 properties above treated differently from these ones?

Reply:
Elements should first be sorted by reference_name, start, end position. Ordering based on the rest of the attributes is not important, but should be deterministic.